phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/common/skin/DetailsListWidgetSkin.java

```diff
@@ -0,0 +1,76 @@
+package org.phoenicis.javafx.components.common.skin;
+
+import javafx.beans.binding.Bindings;
+import javafx.collections.ObservableList;
+import javafx.scene.control.ListView;
+import org.phoenicis.javafx.collections.MappedList;
+import org.phoenicis.javafx.components.common.control.DetailsListElement;
+import org.phoenicis.javafx.components.common.control.DetailsListWidget;
+import org.phoenicis.javafx.views.common.widgets.lists.ListElementListCell;
+import org.phoenicis.javafx.views.common.widgets.lists.ListWidgetEntry;
+import org.phoenicis.javafx.views.common.widgets.lists.ListWidgetSelection;
+
+/**
+ * The skin for the {@link DetailsListWidget} component
+ *
+ * @param <E> The concrete type of the elements shown in this list widget
+ */
+public class DetailsListWidgetSkin<E> extends SkinBase<DetailsListWidget<E>, DetailsListWidgetSkin<E>> {
+    /**
+     * Mapped list between the input {@link ListWidgetEntry} and {@link DetailsListElement}
+     */
+    private final ObservableList<DetailsListElement<E>> mappedElements;
+
+    /**
+     * Constructor
+     *
+     * @param control The control belonging to the skin
+     */
+    public DetailsListWidgetSkin(DetailsListWidget<E> control) {
+        super(control);
+
+        this.mappedElements = new MappedList<>(getControl().getElements(), value -> {
+            DetailsListElement<E> newElement = new DetailsListElement<>(value);
+
+            newElement.setOnMouseClicked(
+                    event -> getControl().setSelectedElement(new ListWidgetSelection<>(value, event)));
+
+            return newElement;
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialise() {
+        final ListView<DetailsListElement<E>> container = new ListView<>();
+        container.getStyleClass().addAll("listWidget", "detailsListWidget");
+
+        container.setPrefWidth(0);
+        container.setPrefHeight(0);
+
+        container.setCellFactory(param -> new ListElementListCell<>());
+
+        // ensure, that updates to the selected element property are automatically reflected in the view
+        getControl().selectedElementProperty().addListener((observable, oldValue, newValue) -> {
+            // deselect the old element
+            if (oldValue != null) {
```

Comment:
It is not the same for all implementations, only for the `DetailsListWidget` and the `CompactListWidget`, because they use a `ListView` underneath.
I'm not sure where to unify the redundant behavior, because I don't want to add two additional classes (a control and a skin class) for two concrete implementations.

Reply:
Ok.